yellowbox/image_build.py

```diff
@@ -46,40 +81,51 @@ def build_image(
     yaspin_spinner = _get_spinner(spinner)
     with yaspin_spinner(f"Creating image {image_tag}..."):
         kwargs = {"tag": image_tag, "rm": True, "forcerm": True, **kwargs}
-        build_log = docker_client.api.build(**kwargs)
-        for msg_b in build_log:
-            msgs = str(msg_b, "utf-8").splitlines()
-            for msg in msgs:
-                try:
-                    parse_msg = json.loads(msg)
-                except JSONDecodeError as e:
-                    raise DockerException("error at build logs") from e
-                s = parse_msg.get("stream")
-                if s and file:
-                    print(s, end="", flush=True, file=file)
-                else:
-                    # runtime errors
-                    error_detail = parse_msg.get("errorDetail")
-                    # parse errors
-                    error_msg = parse_msg.get("message")
-                    # steps of the image creation
-                    status = parse_msg.get("status")
-                    # end of process, will contain the ID of the temporary container created at the end
-                    aux = parse_msg.get("aux")
-                    if error_detail is not None:
-                        raise BuildError(reason=error_detail, build_log=None)
-                    elif error_msg is not None:
-                        raise DockerfileParseError(reason=error_msg, build_log=None)
-                    elif status is not None and file:
-                        print(status, end="", flush=True, file=file)
-                    elif aux is not None and file:
-                        print(aux, end="", flush=True, file=file)
-                    else:
-                        raise DockerException(parse_msg)
+        _docker_build(docker_client, file, **kwargs)
         yield image_tag
         if remove_image:
             try:
                 docker_client.api.remove_image(image_tag)
             except ImageNotFound:
                 # if the image was already deleted
                 pass
+
+
+def _remove_image(docker_client: DockerClient, image_tag: str):
+    docker_client.api.remove_image(image_tag)
+
+
+@asynccontextmanager
+async def async_build_image(
+    docker_client: DockerClient,
+    image_name: str,
+    remove_image: bool = True,
+    file: Optional[TextIO] = sys.stderr,
```

Comment:
having a "file" here is problematic, since it could be very confusing to print progress as it happens (what happens if two images build in parallel?) maybe store all the logs and only print them in case of error/at the end?

Reply:
I added a parameter "async_run" in order to log at the end of the loop